### PR TITLE
Chk 1344 async user receipt

### PR DESCRIPTION
**Pull request by pietro-tota**
Linked PRs:

-[PR](https://github.com/pagopa/pagopa-infra/pull/852) for align transaction statuses on infra
-[PR](https://github.com/pagopa/pagopa-ecommerce-local/pull/62) for align env variables on ecommerce-local

#### List of Changes

1. Make notification process asynchronous. Mail now is sent by event-dispatcher. Transaction service send `TransactionUserReceiptRequestedEvent` with send payment result informations such as outcome, payment date etc so that event dispatcher can send email asynchronously
2. Removed all reference to Notifications service since now email is sent by event-dispatcher
3. Add new state `NOTIFICATION_REQUESTED` that models a transaction for which a user receipt notification has been requested

#### Motivation and Context
Those changes are needed in order to centralize notification send process into event-dispatcher that will take care also of retry in case of missing communication with notification service

#### How Has This Been Tested?
Junit test + local tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -36,13 +32,12 @@ AFM_CONNECTION_TIMEOUT=10000
 AFM_KEY=key
 
 ECOMMERCE_STORAGE_CONNECTION_STRING="DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;QueueEndpoint=http://storage:10001/devstoreaccount1"
-TRANSACTION_AUTHORIZATION_REQUESTED_EVENT_QUEUE_NAME=transactionauthrequestedtevents
-TRANSACTION_AUTHORIZATION_REQUESTED_EVENT_QUEUE_VISIBILITY_TIMEOUT=60
 
 TRANSACTION_EXPIRATION_QUEUE_NAME=pagopa-ecommerce-transactions-expiration-queue
-TRANSACTION_CLOSEPAYMENT_RETRY_QUEUE_NAME=pagopa-ecommerce-transactions-closepayment-retry-queue
+TRANSACTION_CLOSE_PAYMENT_RETRY_QUEUE_NAME=pagopa-ecommerce-transactions-close-payment-retry-queue
```
Comment (Fernando-Granato) [inline .env.example:37]: maybe it should also be updated ecommerce-local with these changes absolutely correct.
What do you think? 🙂

Comment (pietro-tota) [reply .env.example:37]: Done into this [PR](https://github.com/pagopa/pagopa-ecommerce-local/pull/62) (as linked in description section)

Comment (Fernando-Granato) [reply .env.example:37]: Sorry 🙂

**Review by Fernando-Granato — APPROVED** [review]

**Review by Fernando-Granato — APPROVED** [review]
well done 👍

**Review by Fernando-Granato — APPROVED** [review]

**Pull request merged**